ccl/changefeedccl: TestChangefeedTruncateOrDrop failed

ccl/changefeedccl.TestChangefeedTruncateOrDrop [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/14039461?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/14039461?buildTab=artifacts#/) on release-23.1 @ [0c48ab97d8935fa4360a70803c80c1cef71fd958](https://github.com/cockroachdb/cockroach/commits/0c48ab97d8935fa4360a70803c80c1cef71fd958):


```
=== RUN   TestChangefeedTruncateOrDrop
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/905465f697dc04c39f7d8a82cb057bfe/logTestChangefeedTruncateOrDrop2571105095
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestChangefeedTruncateOrDrop
    changefeed_test.go:4445: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/905465f697dc04c39f7d8a82cb057bfe/logTestChangefeedTruncateOrDrop2571105095
--- FAIL: TestChangefeedTruncateOrDrop (13.67s)
=== RUN   TestChangefeedTruncateOrDrop/kafka
    helpers_test.go:823: making server as secondary tenant
    helpers_test.go:901: making kafka feed factory
    changefeed_test.go:4438: expected "\"drop\" was dropped|CHANGEFEED cannot target offline table: drop" error, instead got: descriptor is being dropped
        (1) attached stack trace
          -- stack trace:
          | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*jobFeed).FetchTerminalJobErr
          | 	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/testfeed_test.go:529
          | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.newDepInjector.func2.1
          | 	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/testfeed_test.go:694
          | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*reportErrorResumer).OnFailOrCancel
          | 	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/testfeed_test.go:354
          | github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).stepThroughStateMachine.func3
          | 	github.com/cockroachdb/cockroach/pkg/jobs/registry.go:1767
          | github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).stepThroughStateMachine
          | 	github.com/cockroachdb/cockroach/pkg/jobs/registry.go:1768
          | github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).stepThroughStateMachine
          | 	github.com/cockroachdb/cockroach/pkg/jobs/registry.go:1713
          | github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).runJob
          | 	github.com/cockroachdb/cockroach/pkg/jobs/adopt.go:474
          | github.com/cockroachdb/cockroach/pkg/jobs.(*StartableJob).Start.func2
          | 	github.com/cockroachdb/cockroach/pkg/jobs/jobs.go:978
          | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2
          | 	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:470
          | runtime.goexit
          | 	src/runtime/asm_amd64.s:1594
        Wraps: (2) descriptor is being dropped
        Error types: (1) *withstack.withStack (2) *errutil.leafError
    --- FAIL: TestChangefeedTruncateOrDrop/kafka (13.66s)
```

Parameters:
 - <code>TAGS=bazel,gss,race</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/cdc
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedTruncateOrDrop.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-36150

--- a/pkg/ccl/changefeedccl/changefeed_test.go
+++ b/pkg/ccl/changefeedccl/changefeed_test.go
@@ -59,6 +59,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/server/telemetry"
 	"github.com/cockroachdb/cockroach/pkg/settings/cluster"
 	"github.com/cockroachdb/cockroach/pkg/sql"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog"
 	"github.com/cockroachdb/cockroach/pkg/sql/catalog/desctestutils"
 	"github.com/cockroachdb/cockroach/pkg/sql/execinfra"
 	"github.com/cockroachdb/cockroach/pkg/sql/execinfrapb"
@@ -4428,7 +4429,6 @@ func TestChangefeedUpdatePrimaryKey(t *testing.T) {
 func TestChangefeedTruncateOrDrop(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
-	skip.WithIssue(t, 119328, "flaky test")
 
 	assertFailuresCounter := func(t *testing.T, m *Metrics, exp int64) {
 		t.Helper()
@@ -4485,7 +4485,13 @@ func TestChangefeedTruncateOrDrop(t *testing.T) {
 		defer closeFeed(t, drop)
 		assertPayloads(t, drop, []string{`drop: [1]->{"after": {"a": 1}}`})
 		sqlDB.Exec(t, `DROP TABLE drop`)
-		const dropOrOfflineRE = `"drop" was dropped|CHANGEFEED cannot target offline table: drop`
+		// Dropping the table should cause the schema feed to return an error.
+		// This error can either come from validateDescriptor (the first two)
+		// or the lease manager (catalog.ErrDescriptorDropped).
+		dropOrOfflineRE := fmt.Sprintf(
+			`"drop" was dropped|CHANGEFEED cannot target offline table: drop|%s`,
+			catalog.ErrDescriptorDropped,
+		)
 		if err := drainUntilErr(drop); !testutils.IsError(err, dropOrOfflineRE) {
 			t.Errorf(`expected %q error, instead got: %+v`, dropOrOfflineRE, err)
 		}

